compiler/src/dotty/tools/dotc/core/TypeErasure.scala

```diff
@@ -533,7 +533,7 @@ class TypeErasure(isJava: Boolean, semiEraseVCs: Boolean, isConstructor: Boolean
         // but potentially re-introduced by ResolveSuper, when we add
         // forwarders to mixin methods.
         // See doc comment for ElimByName for speculation how we could improve this.
-      else MethodType(Nil, Nil, eraseResult(sym.info.finalResultType))
+      else MethodType(Nil, Nil, eraseResult(sym.info.finalResultType.underlyingIfRepeated(isJava)))
```

Comment:
It would be good to check if we can move it earlier to simply the contract between phases.

Reply:
That would also include `val` binding generated for repeated parameters.